src/client/sandbox/shadow-ui.js

```diff
@@ -64,6 +64,32 @@ export default class ShadowUI extends SandboxBase {
         return ShadowUI._filterList(styleSheetList, item => ShadowUI._filterElement(item.ownerNode));
     }
 
+    static _filterFirstNonShadowElement (nodeList) {
+        for (var i = 0; i < nodeList.length; i++) {
+            var el = ShadowUI._filterElement(nodeList[i]);
+
+            if (el)
+                return nodeList[i];
+        }
+
+        return null;
+    }
+
+    static _createWrapperForQuerySelectorFn (nativeQuerySelectorFnName, nativeQuerySelectorAllFnName) {
```

Comment:
Lets' pass native functions instead of string names. It allows to write a little more readable code.

Also let's create structure that will contain all `createWrapperFor***` functions.
```js
//declaration

this.createWrapperFor.querySelector = ...
// This will be added later - https://github.com/DevExpress/testcafe-hammerhead/issues/1159
this.createWrapperFor.getElementsByTagName = ...
this.createWrapperFor.getElementsByClassName = ...
....

//usage

docProto.querySelector = this.createWrapperFor.querySelector(nativeMethods.querySelector, nativeMethods.querySelectorAll)
```